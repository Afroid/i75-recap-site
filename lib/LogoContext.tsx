"use client";
import { createContext, useContext, useMemo } from "react";

const logos = [
  "/default-logo.png",
  "/default-logo-alt.png",
  "/default-RWB.png",
];

const LogoContext = createContext<string | null>(null);

export function LogoProvider({ children }: { children: React.ReactNode }) {
  // Randomizes only once on initial render
  const logo = useMemo(
    () => logos[Math.floor(Math.random() * logos.length)],
    []
  );
  return (
    <LogoContext.Provider value={logo}>
      {children}
    </LogoContext.Provider>
  );
}

export function useLogo() {
  const logo = useContext(LogoContext);
  if (logo === null) throw new Error("useLogo must be inside LogoProvider");
  return logo;
}
