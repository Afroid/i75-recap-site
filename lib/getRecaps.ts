import recap2024 from "@/data/recaps/2024.json";
import recap2023 from "@/data/recaps/2023.json";
import { RecapData } from "@/types/types";

export function getRecaps(): Record<number, number[]> {
  const allData: RecapData[] = [recap2024, recap2023]; // Add more as needed
  const summary: Record<number, number[]> = {};

  for (const season of allData) {
    summary[season.year] = season.recaps.map((r) => r.week);
  }

  return summary;
}
