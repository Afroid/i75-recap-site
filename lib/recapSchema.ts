import { z } from "zod";

const RecapMatchupSchema = z.object({
  team1: z.string(),
  team2: z.string(),
  winner: z.string(),
  score: z.string(),
  breakdown: z.string(),
  imageUrl: z.string().optional(),
  imageAlt: z.string().optional(),
});

const GameNotesSectionSchema = z.object({
  type: z.literal("gameNotes"),
  pointsLeader: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  dud: z.object({
    name: z.string(),
    points: z.number(),
    notes: z.string().optional(),
  }),
  matchups: z.array(RecapMatchupSchema),
});

const RecapSectionSchema = z.discriminatedUnion("type", [
  z.object({
    type: z.literal("intro"),
    content: z.string(),
  }),
  z.object({
    type: z.literal("tidbits"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("injuries"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("sidePieces"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("matchups"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  }),
  z.object({
    type: z.literal("farewell"),
    intro: z.string().optional(),
    bullets: z.array(z.string()).optional(),
    outro: z.string().optional(),
    imageUrl: z.string().optional(),
    imageAlt: z.string().optional(),
  })
]);

const RecapWeekSchema = z.object({
  week: z.number(),
  title: z.string(),
  sections: z.array(z.union([RecapSectionSchema, GameNotesSectionSchema])),
});

export const RecapDataSchema = z.object({
  year: z.number(),
  recaps: z.array(RecapWeekSchema),
});
