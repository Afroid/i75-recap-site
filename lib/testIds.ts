export const TestIds = {
  /**
   * Pages
   */

  // Index - App Level
  SITE_HEADER: "site-header",
  MAIN_CONTENT: "main-content",
  PAGE_WRAPPER: "page-wrapper",

  // Home Page
  HOME_MAIN:   "home-main",
  HOME_HERO_LOGO: "home-hero-logo",
  HOME_INTRO_SECTION: "home-intro-section",
  HOME_HERO_HEADING: "home-hero-heading",
  HOME_CTA_CONTAINER: "home-cta-container",
  HOME_VIEW_ALL_RECAPS_BUTTON: "home-view-all-recaps-button",
  HOME_LEARN_MORE_BUTTON: "home-learn-more-button",

  // 404 Page
  NOT_FOUND_MAIN: "not-found-main",
  RETURN_HOME_LINK: "return-home-link",

  // About Page
  ABOUT_MAIN: "about-main",

  // Contact Page
  CONTACT_MAIN: "contact-main",
  CONTACT_EMAIL_LINK: "contact-email-link",
  CONTACT_FORM: "contact-form",
  CONTACT_FORM_NAME_SECTION: "contact-form-name-section",
  CONTACT_FORM_EMAIL_SECTION: "contact-form-email-section",
  CONTACT_FORM_MESSAGE_SECTION: "contact-form-message-section",
  CONTACT_FORM_SUBMIT: "contact-form-submit",

  /**
   * Recap Pages - (pages/recaps)
   */

  // Recap Page

  // View All Recaps Page
  VIEW_ALL_RECAPS: "view-all-recaps",
  // Year Page

  /**
   * Components
   */

  // Breadcrumb Component
  BREADCRUMB: "breadcrumb",
  BREADCRUMB_NAV: "breadcrumb-nav",
  BREADCRUMB_LIST: "breadcrumb-list",
  BREADCRUMB_ITEM: "breadcrumb-item",
  BREADRCRUMB_LINK: "breadcrumb-link",
  BREADCRUMB_CURRENT: "breadcrumb-current",
  BREADCRUMB_SEPARATOR: "breadcrumb-separator",

  // DesktopNavbar Component

  // GameNotesSection.tsx
  GAME_NOTES_SECTION: "game-notes-section",
  GAME_NOTES_ITEM: "game-notes-item",
  GAME_NOTES_SCOREBOARD: "game-notes-scoreboard",
  GAME_NOTES_BREAKDOWN: "game-notes-breakdown",
  GAME_NOTES_IMAGE_CONTAINER: "game-notes-image-container",
  GAME_NOTES_IMAGE: "game-notes-image",

  // Header.tsx
  HEADER_SITE_HEADER: "site-header",
  HEADER_MOBILE_HEADER: "mobile-header",
  HEADER_HAMBURGER_BUTTON: "header-hamburger-button",
  HEADER_MOBILE_LOGO: "header-mobile-logo",
  HEADER_DESKTOP_LOGO: "header-desktop-logo",
  HEADER_DESKTOP_NAV: "header-desktop-nav",
  HEADER_MOBILE_DRAWER: "header-mobile-drawer",

  // HeaderLogo.tsx
  HEADER_LOGO_LINK: "header-logo-link",
  HEADER_LOGO_BG_DARK: "header-logo-bg-dark",
  HEADER_LOGO_BG_LIGHT: "header-logo-bg-light",
  HEADER_LOGO_CONTAINER: "header-logo-container",
  HEADER_LOGO_NAME: "header-logo-name",
  HEADER_LOGO_DYNAMIC: "header-logo-dynamic",

  // LeaderDudSection.tsx
  LDS_SECTION: "lds-section",
  LDS_LEADER_HEADING: "lds-leader-heading",
  LDS_LEADER_POINTS: "lds-leader-points",
  LDS_LEADER_NOTES: "lds-leader-notes",
  LDS_DUD_HEADING: "lds-dud-heading",
  LDS_DUD_POINTS: "lds-dud-points",
  LDS_DUD_NOTES: "lds-dud-notes",

  // MobileDrawer.tsx
  MOBILE_SEASON_TOGGLE: "mobile-season-toggle",
  MOBILE_SEASON_WEEKS: "mobile-season-weeks",
  MOBILE_WEEK_LINK: "mobile-week-link",
  MOBILE_DRAWER_OVERLAY: "mobile-drawer-overlay",
  MOBILE_DRAWER: "mobile-drawer",
  MOBILE_DRAWER_TITLE: "mobile-drawer-title",
  MOBILE_OLDER_SEASON_TITLE: "mobile-older-season-title",
  MOBILE_TOGGLE_OLDER_SEASONS: "mobile-toggle-older-seasons",

  // RecapLayout.tsx
  RECAP_TITLE: "recap-title",
  RECAP_SECTION: "recap-section",
  RECAP_HEADING: "recap-heading",
  RECAP_INTRO: "recap-intro",
  RECAP_BULLETS: "recap-bullets",
  RECAP_BULLET: "recap-bullet",
  RECAP_CONTENT: "recap-content",
  RECAP_IMAGE: "recap-image",

  // Scoreboard.tsx
  SCOREBOARD_CONTAINER: "scoreboard-container",
  SCOREBOARD_ENTRY: "scoreboard-entry",
  SCOREBOARD_LOGO: "scoreboard-logo",
  SCOREBOARD_NAME: "scoreboard-name",
  SCOREBOARD_RECORD: "scoreboard-record",
  SCOREBOARD_SCORE: "scoreboard-score",

  // TeamLogo.tsx
  TEAM_LOGO_IMAGE: "team-logo-image",

  // Static Links
  NAV_LINK_HOME: "nav-link-home",
  NAV_LINK_ABOUT: "nav-link-about",
  NAV_LINK_CONTACT: "nav-link-contact"
}
