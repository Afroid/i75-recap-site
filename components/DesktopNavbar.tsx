"use client";

import Link from "next/link";
import { useState, useRef, useEffect } from "react";
import { TestIds } from "@/lib/testIds";

type Summary = Record<string, number[]>;

/**
 * DesktopNavbar
 * Fetches recap summary and renders a hover-triggered flyout menu
 * showing recent seasons and weeks, plus navigation links.
 */
export default function DesktopNavbar() {
  // Stores recap counts per year
  const [recapData, setRecapData] = useState<Summary>({});
  const [isFlyoutOpen, setIsFlyoutOpen] = useState(false);
  const [showAllSeasons, setShowAllSeasons] = useState(false);
  const flyoutRef = useRef<HTMLDivElement>(null);

  // Fetch summary from our API on mount
  useEffect(() => {
    fetch("/api/recapsSummary")
      .then((r) => r.json())
      .then(setRecapData)
      .catch(console.error);
  }, []);

  // Close flyout when clicking outside
  useEffect(() => {
    function handleClickOutside(event: MouseEvent) {
      if (
        flyoutRef.current &&
        !flyoutRef.current.contains(event.target as Node)
      ) {
        setIsFlyoutOpen(false); // Clicked outside â†’ close flyout
      }
    }

    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []);


  // Sorts all of the seasons in descending order
  const sortedSeasons = Object.entries(recapData).sort((a, b) => Number(b[0]) - Number(a[0]));

  // Split seasons from the last three versus the older ones after that.
  const latestSeasons = sortedSeasons.slice(0, 3); // Only latest 3 seasons
  const olderSeasons = sortedSeasons.slice(3);

  return (
    <div data-testid={TestIds.DESKTOP_NAVBAR} className="flex items-center justify-between w-full">
      {/* Left side: Recaps */}
      <div
        data-testid={TestIds.DESKTOP_NAV_RECAPS}
        className="relative flex flex-col items-center group w-24"
        onMouseEnter={() => setIsFlyoutOpen(true)}
      >
        <button
          data-testid={TestIds.DESKTOP_RECAPS_BUTTON}
          className={[
            "w-full h-12 flex",
            "items-center justify-center",
            "hover:text-green-400 transition-colors duration-300"
          ].join(" ")}
        >
          Recaps
        </button>

        {/* Pointer under Recaps */}
        <div
          className={[
            "absolute top-[calc(100%-.5rem)] w-4 h-4",
            "bg-white rotate-45 shadow-lg",
            "hidden group-hover:block"
          ].join(" ")}
        />

        {/* Flyout menu */}
        {isFlyoutOpen && (
          <div
            data-testid={TestIds.DESKTOP_RECAPS_FLYOUT}
            ref={flyoutRef}
            className={[
              "absolute top-full",
              "flex-col",
              "bg-gray-50 text-black",
              "shadow-lg rounded-md",
              "z-50 min-w-[180px] py-2",
              "ease-in-out"
            ].join(" ")}
          >
            {/* Latest Seasons */}
            {latestSeasons.map(([year, weeks]) => (
              <div
                key={year}
                data-testid={`${TestIds.DESKTOP_SEASON}-${year}`}
                className="relative group/season w-full"
              >
                <button
                  data-testid={`${TestIds.DESKTOP_SEASON_BUTTON}-${year}`}
                  className={[
                    "w-full flex",
                    "justify-center items-center",
                    "h-10 px-4 py-2",
                    "rounded-md whitespace-nowrap",
                    "hover:bg-gradient-to-r hover:from-green-100 hover:to-green-300"
                  ].join(" ")}
                >
                  {year} Season
                </button>

                {/* Sub-Flyout Pointer */}
                <div
                  className={[
                    "absolute top-1/2 left-full -translate-y-1/2 -ml-2 w-4 h-4",
                    "bg-gray-50 shadow-md",
                    "rotate-45 hidden",
                    "group-hover/season:block"
                  ].join(" ")}
                />

                {/* Sub-Flyout: Weeks */}
                <div
                  className={[
                    "absolute top-0 left-full",
                    "hidden group-hover/season:flex flex-col",
                    "bg-gray-50 text-black",
                    "p-2 min-w-[100px] w-full",
                    "rounded-md"
                  ].join(" ")}
                >
                  {weeks.map((week) => (
                    <Link
                      key={week}
                      href={`/recaps/${year}/week-${week}`}
                      data-testid={`${TestIds.DESKTOP_WEEK}-${year}-${week}`}
                      onClick={() => setIsFlyoutOpen(false)}
                      className={[
                        "px-4 py-2",
                        "rounded-md",
                        "hover:bg-gradient-to-r hover:from-green-300 hover:to-green-100",
                        "transition-colors duration-300 whitespace-nowrap"
                      ].join(" ")}
                    >
                      Week {week}
                    </Link>
                  ))}
                </div>
              </div>
            ))}

            {/* Older Seasons - Initially, these are hidden. */}
            {showAllSeasons && olderSeasons.length > 0 && (
              <>
                <div className="border-t border-gray-300 my-2" />
                <h2
                  className={[
                    "font-bold text-lg text-center text-green-600 text-decoration-line: underline",
                    "mb-2"
                  ].join(" ")}
                >
                  Older Recaps
                </h2>
                {olderSeasons.map(([year, weeks]) => (
                  <div
                    key={year}
                    data-testid={`${TestIds.DESKTOP_OLDER_SEASON}-${year}`}
                    className="relative group/season w-full"
                  >
                    <button
                      data-testid={`${TestIds.DESKTOP_OLDER_SEASON_BUTTON}-${year}`}
                      className={[
                        "flex justify-center w-full items-center",
                        "h-10 px-4 py-2",
                        "rounded-md",
                        "hover:bg-gradient-to-r hover:from-green-100 hover:to-green-300",
                        "transition-colors duration-300 whitespace-nowrap"
                      ].join(" ")}
                    >
                      {year} Season
                    </button>

                    {/* Sub-Flyout Pointer */}
                    <div
                      className={[
                        "absolute top-1/2 left-full -translate-y-1/2",
                        "-ml-2 w-4 h-4",
                        "bg-gray-50 rotate-45 hidden group-hover/season:block shadow-md"
                      ].join(" ")}
                    />

                    {/* Sub-Flyout: Weeks */}
                    <div
                      className={[
                        "absolute top-0 left-full",
                        "hidden group-hover/season:flex flex-col",
                        "bg-gray-50 text-black",
                        "rounded-md p-2 min-w-[100px] w-full"
                    ].join(" ")}
                    >
                      {weeks.map((week) => (
                        <Link
                          key={week}
                          href={`/recaps/${year}/week-${week}`}
                          onClick={() => setIsFlyoutOpen(false)}
                          className={[
                            "px-4 py-2",
                            "hover:bg-gradient-to-r hover:from-green-300 hover:to-green-100",
                            "rounded-md transition-colors duration-300",
                            "whitespace-nowrap"
                          ].join(" ")}
                        >
                          Week {week}
                        </Link>
                      ))}
                    </div>
                  </div>
                ))}
              </>
            )}

            {/* Divider */}
            <div className="border-t border-gray-300 my-2" />

            {/* View Older Seasons */}
            {olderSeasons.length > 0 && (
              <button
                data-testid={TestIds.DESKTOP_TOGGLE_OLDER_SEASONS}
                onClick={(e) => {
                  e.stopPropagation(); // This prevents mousedown from reaching document
                  setShowAllSeasons((prev) => !prev);
                }}
                className={[
                  "block text-center w-full",
                  "px-4 py-2",
                  "text-green-600 hover:text-green-800",
                  "font-semibold hover:bg-green-300",
                  "rounded-md transition-colors duration-300"
                ].join(" ")}
              >
                {showAllSeasons ? 'Hide Older Seasons' : 'View Older Seasons'}
              </button>
            )}

            {/* View All Recaps */}
            <Link
              href="/recaps/viewAllRecaps"
              data-testid={TestIds.VIEW_ALL_RECAPS}
              className={[
                "block text-center",
                "px-4 py-2",
                "text-blue-600 hover:text-blue-800 font-semibold hover:bg-blue-300",
                "rounded-md transition-colors duration-300"
              ].join(" ")}
              onClick={() => setIsFlyoutOpen(false)}
            >
              View All Recaps
            </Link>

          </div>
        )}
      </div>

      {/* Right side: Static Navigation Links */}
      <div data-testid="nav-static-links" className="flex items-center space-x-8 ml-auto">
        <Link
          data-testid={TestIds.NAV_LINK_HOME}
          href="/"
          className="hover:text-green-400 transition-colors duration-300">
            Home
        </Link>
        <Link
          data-testid={TestIds.NAV_LINK_ABOUT}
          href="/about"
          className="hover:text-green-400 transition-colors duration-300"
        >
          About
        </Link>
        <Link
          data-testid={TestIds.NAV_LINK_CONTACT}
          href="/contact"
          className="hover:text-green-400 transition-colors duration-300"
        >
          Contact
        </Link>
      </div>
    </div>
  );
}
