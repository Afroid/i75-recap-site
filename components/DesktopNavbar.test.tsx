import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import DesktopNavbar from "./DesktopNavbar";
import { TestIds } from "@/lib/testIds";

describe("DesktopNavbar", () => {
  const fakeSummary = {
    "2023": [1, 2, 3],
    "2022": [5],
    "2021": [4],
    "2020": [6],
  };

  beforeEach(() => {
    // Stub global.fetch to return our fake summary
    global.fetch = jest.fn().mockResolvedValue({
      json: async () => fakeSummary,
    });
  });

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("fetches data and displays latest seasons on hover, and allows showing older seasons",
    async () => {
      render(<DesktopNavbar />);

      // Expect fetch to have been called
      await waitFor(() => expect(global.fetch).toHaveBeenCalledWith("/api/recapsSummary"));

      // Hover over the recaps area to open the flyout
      fireEvent.mouseEnter(screen.getByTestId(TestIds.DESKTOP_NAV_RECAPS));

      // Wait until season buttons are rendered
      await waitFor(() => expect(screen.getByTestId(TestIds.DESKTOP_SEASON + "-2023"))
        .toBeInTheDocument());
      expect(screen.getByTestId(TestIds.DESKTOP_SEASON_BUTTON + "-2023"))
        .toHaveTextContent("2023 Season");
      expect(screen.getByTestId(TestIds.DESKTOP_SEASON_BUTTON + "-2022"))
        .toHaveTextContent("2022 Season");
      expect(screen.getByTestId(TestIds.DESKTOP_SEASON_BUTTON + "-2021"))
        .toHaveTextContent("2021 Season");

      // Older seasons should not be visible until toggled
      expect(screen.queryByTestId(TestIds.DESKTOP_OLDER_SEASON + "-2020")).toBeNull();

      // Toggle older seasons
      fireEvent.click(screen.getByTestId(TestIds.DESKTOP_TOGGLE_OLDER_SEASONS));
      expect(screen.getByTestId(TestIds.DESKTOP_OLDER_SEASON + "-2020")).toBeInTheDocument();
      expect(screen.getByTestId(TestIds.DESKTOP_OLDER_SEASON_BUTTON + "-2020"))
        .toHaveTextContent("2020 Season");
  });

  it("closes the flyout when clicking outside", async () => {
    // Mock out fetch so the component actually renders something, even something fake
    const fakeSummary = { "2023": [1], "2022": [2] };

    (global.fetch as jest.Mock) = jest.fn().mockResolvedValue({
      json: async () => fakeSummary,
    });

    // Render and wait for the initial fetch
    render(<DesktopNavbar />);
    await waitFor(() => expect(global.fetch).toHaveBeenCalledWith("/api/recapsSummary"));

    // Hover over the recaps area to open the flyout
    fireEvent.mouseEnter(screen.getByTestId(TestIds.DESKTOP_NAV_RECAPS));
    // Confirm it opened
    expect(await screen.findByTestId("desktop-recaps-flyout")).toBeInTheDocument();

    // Simulate a click happened outside of the flyout container
    fireEvent.mouseDown(document.body);

    // Flyout should now be gone
    await waitFor(() => {
      expect(screen.queryByTestId("desktop-recaps-flyout")).toBeNull();
    });
  });


  it("matches snapshot of the flyout with latest seasons", async () => {
    const { asFragment } = render(<DesktopNavbar />);
    await waitFor(() => expect(global.fetch).toHaveBeenCalled());

    fireEvent.mouseEnter(screen.getByTestId(TestIds.DESKTOP_NAV_RECAPS));
    await waitFor(() => expect(screen.getByTestId(TestIds.DESKTOP_SEASON + "-2023"))
      .toBeInTheDocument());

    expect(asFragment()).toMatchSnapshot();
  });
});
