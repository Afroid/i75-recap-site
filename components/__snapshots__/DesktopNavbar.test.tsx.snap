// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DesktopNavbar matches snapshot of the flyout with latest seasons 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between w-full"
    data-testid="desktop-navbar"
  >
    <div
      class="relative flex flex-col items-center group w-24"
      data-testid="desktop-nav-recaps"
    >
      <button
        class="w-full h-12 flex items-center justify-center hover:text-green-400 transition-colors duration-300"
        data-testid="desktop-recaps-button"
      >
        Recaps
      </button>
      <div
        class="absolute top-[calc(100%-.5rem)] w-4 h-4 bg-white rotate-45 shadow-lg hidden group-hover:block"
      />
      <div
        class="absolute top-full flex-col bg-gray-50 text-black shadow-lg rounded-md z-50 min-w-[180px] py-2 ease-in-out"
        data-testid="desktop-recaps-flyout"
      >
        <div
          class="relative group/season w-full"
          data-testid="desktop-season-2023"
        >
          <button
            class="w-full flex justify-center items-center h-10 px-4 py-2 rounded-md whitespace-nowrap hover:bg-gradient-to-r hover:from-green-100 hover:to-green-300"
            data-testid="desktop-season-button-2023"
          >
            2023 Season
          </button>
          <div
            class="absolute top-1/2 left-full -translate-y-1/2 -ml-2 w-4 h-4 bg-gray-50 shadow-md rotate-45 hidden group-hover/season:block"
          />
          <div
            class="absolute top-0 left-full hidden group-hover/season:flex flex-col bg-gray-50 text-black p-2 min-w-[100px] w-full rounded-md"
          >
            <a
              class="px-4 py-2 rounded-md hover:bg-gradient-to-r hover:from-green-300 hover:to-green-100 transition-colors duration-300 whitespace-nowrap"
              data-testid="desktop-week-2023-1"
              href="/recaps/2023/week-1"
            >
              Week 1
            </a>
            <a
              class="px-4 py-2 rounded-md hover:bg-gradient-to-r hover:from-green-300 hover:to-green-100 transition-colors duration-300 whitespace-nowrap"
              data-testid="desktop-week-2023-2"
              href="/recaps/2023/week-2"
            >
              Week 2
            </a>
            <a
              class="px-4 py-2 rounded-md hover:bg-gradient-to-r hover:from-green-300 hover:to-green-100 transition-colors duration-300 whitespace-nowrap"
              data-testid="desktop-week-2023-3"
              href="/recaps/2023/week-3"
            >
              Week 3
            </a>
          </div>
        </div>
        <div
          class="relative group/season w-full"
          data-testid="desktop-season-2022"
        >
          <button
            class="w-full flex justify-center items-center h-10 px-4 py-2 rounded-md whitespace-nowrap hover:bg-gradient-to-r hover:from-green-100 hover:to-green-300"
            data-testid="desktop-season-button-2022"
          >
            2022 Season
          </button>
          <div
            class="absolute top-1/2 left-full -translate-y-1/2 -ml-2 w-4 h-4 bg-gray-50 shadow-md rotate-45 hidden group-hover/season:block"
          />
          <div
            class="absolute top-0 left-full hidden group-hover/season:flex flex-col bg-gray-50 text-black p-2 min-w-[100px] w-full rounded-md"
          >
            <a
              class="px-4 py-2 rounded-md hover:bg-gradient-to-r hover:from-green-300 hover:to-green-100 transition-colors duration-300 whitespace-nowrap"
              data-testid="desktop-week-2022-5"
              href="/recaps/2022/week-5"
            >
              Week 5
            </a>
          </div>
        </div>
        <div
          class="relative group/season w-full"
          data-testid="desktop-season-2021"
        >
          <button
            class="w-full flex justify-center items-center h-10 px-4 py-2 rounded-md whitespace-nowrap hover:bg-gradient-to-r hover:from-green-100 hover:to-green-300"
            data-testid="desktop-season-button-2021"
          >
            2021 Season
          </button>
          <div
            class="absolute top-1/2 left-full -translate-y-1/2 -ml-2 w-4 h-4 bg-gray-50 shadow-md rotate-45 hidden group-hover/season:block"
          />
          <div
            class="absolute top-0 left-full hidden group-hover/season:flex flex-col bg-gray-50 text-black p-2 min-w-[100px] w-full rounded-md"
          >
            <a
              class="px-4 py-2 rounded-md hover:bg-gradient-to-r hover:from-green-300 hover:to-green-100 transition-colors duration-300 whitespace-nowrap"
              data-testid="desktop-week-2021-4"
              href="/recaps/2021/week-4"
            >
              Week 4
            </a>
          </div>
        </div>
        <div
          class="border-t border-gray-300 my-2"
        />
        <button
          class="block text-center w-full px-4 py-2 text-green-600 hover:text-green-800 font-semibold hover:bg-green-300 rounded-md transition-colors duration-300"
          data-testid="desktop-toggle-older-seasons"
        >
          View Older Seasons
        </button>
        <a
          class="block text-center px-4 py-2 text-blue-600 hover:text-blue-800 font-semibold hover:bg-blue-300 rounded-md transition-colors duration-300"
          data-testid="view-all-recaps"
          href="/recaps/viewAllRecaps"
        >
          View All Recaps
        </a>
      </div>
    </div>
    <div
      class="flex items-center space-x-8 ml-auto"
      data-testid="nav-static-links"
    >
      <a
        class="hover:text-green-400 transition-colors duration-300"
        data-testid="nav-link-home"
        href="/"
      >
        Home
      </a>
      <a
        class="hover:text-green-400 transition-colors duration-300"
        data-testid="nav-link-about"
        href="/about"
      >
        About
      </a>
      <a
        class="hover:text-green-400 transition-colors duration-300"
        data-testid="nav-link-contact"
        href="/contact"
      >
        Contact
      </a>
    </div>
  </div>
</DocumentFragment>
`;
